system/security/cryptohelper/pke.cpp

```diff
@@ -73,14 +73,27 @@ void CLoadedKey::finalize(RSA *_rsa, const char *_keyName)
     keyName.set(_keyName);
 }
 
+//called during PEM_read_bio_RSA_PUBKEY to set passphrase
+int passphraseCB(char *passPhraseBuf, int passPhraseBufSize, int rwflag, void *pPassPhraseMB)
+{
+    size32_t len = ((MemoryBuffer*)pPassPhraseMB)->length();
+    if (passPhraseBufSize >= (int)len)
```

Comment:
why would it not always be an exact match? If mismatched would indicate a problem wouldn't?

Reply:
From what I can tell passPhraseBufSize doesn't reflect the size of our mb, it's an internal buff which looks to be set to PEM_BUFSIZE  which is 1024 at least at the location I looked.  
We should probably make sure provided passphrases do not exceed this value early on, and give an error, rather than failing here?

Reply:
FYI Openssl source code is pretty darn obfuscated at least at the top level.

Reply:
Yes @afishbeck, I cloned the OpenSSL repo a while back, and the only thing more unreadable than it is the OpenSSL API reference manual. This is the right place to check that the passphrase isn't too long, since we don't know what size OpenSSL passphrase supports until this point in the code

Reply:
Shouldn't this be '>' not '>=' ?

Reply:
@RussWhitehead - what is the verdict on this?

Reply:
If it is equal then it is big enough for the passphrase